### ISSUE TITLE
Proposal: Remove _Ptr(_Const) typedefs

### ISSUE THREAD

**Issue by glebm** [sass/libsass#2785]
The libsass codebase uses typedefs for pointers and const pointers, such as `Expression_Ptr`.

However, pointers should be handled with care but it's easy to miss a pointer if it doesn't look like one.

@xzyfer @mgreter What do you think about removing the `_Ptr` and `_Ptr_Const` typedefs?

This can be done with the following command (GNU sed, for macOS and FreeBSD use `sed -i ''`):

```bash
find src \( -name '*.cpp' -o -name '*.hpp' \) -exec sed -i \
'/typedef.*_Ptr[_;]/d;'\
's/\([A-Za-z_]*\)_Ptr_Const/const \1*/g;'\
's/\([A-Za-z_]*\)_Ptr/\1*/g;'\
's/\([a-z]\)##\*/\1*/g' '{}' \+
```

**Comment by xzyfer** [comment]
I am fully in favour of removing these. I continually struggle to know what types I'm dealing with at any given time.

**Comment by glebm** [comment]
Great! I'm going to wait until @mgreter finishes #2557 to avoid merge conflicts.

**Comment by mgreter** [comment]
I'm not really pro or con this change. My initial reason to introduce these types was to be able to check (search) which types I already converted during the conversion from global memory handling to reference counted memory handling. What I don't like are the `Const_Ptr` types, but I didn't find another solution to solve the issues at hand. And I guess it's rather cosmetics and personal choice what one prefers.

@xzyfer not sure I got your comment right, but the difference is pretty easy. `*_Obj` is something that is reference counted and will be deleted when it goes out of scope, a `*_Ptr` is just a pointer to that object and it will not automatically be cleaned up if it is not assigned to an `*_Obj` later on ...
